src/pypi_attestation_models/_cli.py

```
@@ -0,0 +1,302 @@
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    return parser
+
+
+def _die(args: argparse.Namespace, message: str) -> NoReturn:
+    """Handle argument parsing errors and terminate the program.
+
+    Fix up the type hints on our use of `ArgumentParser.error`.
+    """
+    args._parser.error(message)  # noqa: SLF001.
+    raise ValueError("unreachable")
+
+
+def get_identity_token(args: argparse.Namespace) -> IdentityToken:
+    """Generate an Identity Token.
+
+    This method uses the following order of precedence:
+    - A token passed as an argument
+    - An ambient credential
+    - An OAuth-2 flow
+    """
+    # First, check if a token was supplied
+    if args.identity_token:
+        return IdentityToken(args.identity_token)
+
+    # Ambient credential detection
+    oidc_token = detect_credential()
+    if oidc_token is not None:
+        return IdentityToken(oidc_token)
+
+    # Finally, OAuth-2 Flow
+    if args.staging:
+        issuer = Issuer.staging()
+    else:
+        issuer = Issuer.production()
+
+    return issuer.identity_token()
+
+
+def _sign(args: argparse.Namespace) -> None:
+    """Sign the files passed as argument."""
+    try:
+        identity = get_identity_token(args)
+    except IdentityError as identity_error:
+        _die(args, f"Failed to detect identity: {identity_error}")
+
+    if args.staging:
+        signing_ctx = SigningContext.staging()
+    else:
+        signing_ctx = SigningContext.production()
+
+    with signing_ctx.signer(identity, cache=True) as signer:
+        for file_path in args.files:
+            _logger.debug(f"Signing {file_path}")
+
+            if not file_path.is_file():
+                _die(args, f"{file_path} is not a file.")
+
+            signature_path = Path(f"{file_path}.publish.attestation")
+            if signature_path.is_file():
+                _die(args, f"Signature already exists for {file_path}")
```

Same here (fail early if any of the attestation files already exists)